packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/notifications/RealmDictionaryNotificationsTests.kt

```diff
@@ -0,0 +1,390 @@
+                    assertEquals(dataSet.size - 1, insertions.size)
+                    dataSet.map { it.first }
+                        .also { keys ->
+                            keys.containsAll(insertions.toList())
+                        }
+                }
+                assertEquals(0, dictionaryChange.deletions.size)
+            }
+
+            // Assert notification on removal of elements
+            realm.writeBlocking {
+                val queriedContainer = findLatest(container)!!
+                val queriedDictionary = queriedContainer.nullableObjectDictionaryField
+                queriedDictionary.remove(dataSet[0].first)
+            }
+
+            channel.receive().let { dictionaryChange ->
+                assertIs<UpdatedDictionary<*>>(dictionaryChange)
+
+                assertNotNull(dictionaryChange.map)
+                assertEquals(dataSet.size - 1, dictionaryChange.map.size)
+                assertEquals(1, dictionaryChange.deletions.size)
+                assertEquals(0, dictionaryChange.insertions.size)
+            }
+
+            // Assert notification on removal of elements via values iterator
+            realm.writeBlocking {
+                val queriedContainer = findLatest(container)!!
+                val queriedDictionary = queriedContainer.nullableObjectDictionaryField
+                val iterator = queriedDictionary.values.iterator()
+                iterator.next()
+                iterator.remove()
+            }
+
+            channel.receive().let { dictionaryChange ->
+                assertIs<UpdatedDictionary<*>>(dictionaryChange)
+
+                assertNotNull(dictionaryChange.map)
+                assertEquals(dataSet.size - 2, dictionaryChange.map.size)
+                assertEquals(1, dictionaryChange.deletions.size)
+                assertEquals(0, dictionaryChange.insertions.size)
+            }
+
+            // Assert notification on removal of elements via entry set iterator
+            realm.writeBlocking {
+                val queriedContainer = findLatest(container)!!
+                val queriedDictionary = queriedContainer.nullableObjectDictionaryField
+                val iterator = queriedDictionary.entries.iterator()
+                iterator.next()
+                iterator.remove()
+            }
+
+            channel.receive().let { dictionaryChange ->
+                assertIs<UpdatedDictionary<*>>(dictionaryChange)
+
+                assertNotNull(dictionaryChange.map)
+                assertTrue(dictionaryChange.map.isEmpty())
+                assertEquals(1, dictionaryChange.deletions.size)
+                assertEquals(0, dictionaryChange.insertions.size)
+            }
+
+            observer.cancel()
+            channel.close()
+        }
+    }
+
+    @Test
+    override fun cancelAsFlow() {
+        runBlocking {
+            // Freeze values since native complains if we reference a package-level defined variable
```

Comment:
I don't think this is still the case with the new memory model?

Reply:
That's probably true. A clear case of rogue copy-paste 😓